src/network-policy.ts

```diff
@@ -0,0 +1,517 @@
+   */
+  readonly ports?: NetworkPolicyPort[];
+
+  /**
+   * Peer this rule interacts with.
+   */
+  readonly peer: INetworkPolicyPeer;
+
+}
+
+/**
+ * Describes a particular CIDR (Ex. "192.168.1.1/24","2001:db9::/64") that is
+ * allowed to the pods matched by a network policy selector.
+ * The except entry describes CIDRs that should not be included within this rule.
+ */
+export class NetworkPolicyIpBlock implements INetworkPolicyPeer {
+
+  /**
+   * Create an IPv4 peer from a CIDR
+   */
+  public static ipv4(cidrIp: string, except?: string[]): NetworkPolicyIpBlock {
+    const cidrMatch = cidrIp.match(/^(\d{1,3}\.){3}\d{1,3}(\/\d+)?$/);
+
+    if (!cidrMatch) {
+      throw new Error(`Invalid IPv4 CIDR: "${cidrIp}"`);
+    }
+
+    if (!cidrMatch[2]) {
+      throw new Error(`CIDR mask is missing in IPv4: "${cidrIp}". Did you mean "${cidrIp}/32"?`);
+    }
+
+    return new NetworkPolicyIpBlock(cidrIp, except);
+  }
+
+  /**
+   * Any IPv4 address
+   */
+  public static anyIpv4(): NetworkPolicyIpBlock {
+    return new NetworkPolicyIpBlock('0.0.0.0/0');
+  }
+
+  /**
+   * Create an IPv6 peer from a CIDR
+   */
+  public static ipv6(cidrIp: string, except?: string[]): NetworkPolicyIpBlock {
+
+    const cidrMatch = cidrIp.match(/^([\da-f]{0,4}:){2,7}([\da-f]{0,4})?(\/\d+)?$/);
+
+    if (!cidrMatch) {
+      throw new Error(`Invalid IPv6 CIDR: "${cidrIp}"`);
+    }
+
+    if (!cidrMatch[3]) {
+      throw new Error(`CIDR mask is missing in IPv6: "${cidrIp}". Did you mean "${cidrIp}/128"?`);
+    }
+
+    return new NetworkPolicyIpBlock(cidrIp, except);
+  }
+
+  /**
+   * Any IPv6 address
+   */
+  public static anyIpv6(): NetworkPolicyIpBlock {
+    return new NetworkPolicyIpBlock('::/0');
+  }
+
+  private constructor(
+    /**
+     * A string representing the IP Block Valid examples are "192.168.1.1/24" or "2001:db9::/64".
+     */
+    public readonly cidr: string,
+    /**
+     * A slice of CIDRs that should not be included within an IP Block Valid examples are "192.168.1.1/24" or "2001:db9::/64".
+     * Except values will be rejected if they are outside the CIDR range.
+     */
+    public readonly except?: string[]) {}
+
+
+  /**
+   * @see INetworkPolicyPeer.toNetworkPolicyPeerConfig()
+   */
+  public toNetworkPolicyPeerConfig(): NetworkPolicyPeerConfig {
+    return { ipBlock: this };
+  }
+
+  /**
+   * @internal
+   */
+  public _toKube(): k8s.IpBlock {
+    return { cidr: this.cidr, except: this.except };
+  }
+
+}
+
+/**
+ * Network protocols.
+ */
+export enum NetworkProtocol {
+  /**
+   * TCP.
+   */
+  TCP = 'TCP',
+  /**
+   * UDP.
+   */
+  UDP = 'UDP',
+  /**
+   * SCTP.
+   */
+  SCTP = 'SCTP',
+}
+
+/**
+ * Default behaviors of network traffic in policies.
+ */
+export enum NetworkPolicyTrafficDefault {
+  /**
+   * The policy denies all traffic.
+   * Since rules are additive, additional rules or policies can allow
+   * specific traffic.
+   */
+  DENY = 'DENY',
```

Comment:
Consider creating specific constructs for these:

- `AllowAllEgressNetworkPolicy`...
- `AllowAllIngressNetworkPolicy`...

Reply:
Maybe we will add those in the future, if we see some confusion arising around it. 
These new constructs will then simply use the current API.